contracts/staking/SimpleStaking.sol

```diff
@@ -66,6 +66,8 @@ contract SimpleStaking is AbstractGoodStaking, StakingToken {
 		setDAO(_ns);
 		token = ERC20(_token);
 		iToken = ERC20(_iToken);
+		decimals = token.decimals(); // Staking token decimals should be same with token's decimals
+		tokenDecimalDifference = 18 - token.decimals();
```

Comment:
add a test, to make sure this reverts if token.decimals > 18
should be reverted on solidity 8

Reply:
Changed